### PR TITLE
Disabled Features should be logged 

### PR DESCRIPTION
**Pull request by Abhishekk24**
I added a feature to log a message when a certain function is turned off. This log message will help us know which features are disabled and understand why. It's like a notification that tells us, "Hey, we turned off this specific feature, and here's why." This can be useful for keeping track of the configuration and troubleshooting any issues related to disabled features.

### CODE REVIEW

**Comment by Zabuzard** [comment]
@Abhishekk24 Note that the issue ur fixing was already assigned to another user, who also already created a solution at #899 .

So you might have done "double work" here. Ideally, before starting on an issue, you should get yourself assigned to it first - and if its already assigned to someone, coordinate quickly 🙂

**Comment by Abhishekk24** [comment]
Sure will consider this for further work . Thank you.

**Comment by Zabuzard** [comment]
Please fix the issues mentioned by the failing checks, thanks.

**Comment by Abhishekk24** [comment]
yes sure

**Comment by Abhishekk24** [comment]
If there are anymore errors please guide me.. This is all new to me

File: application/src/main/java/org/togetherjava/tjbot/features/Features.java [hunk 1]
```diff
@@ -73,9 +75,11 @@ private Features() {
      * @param jda the JDA instance commands will be registered at
      * @param database the database of the application, which features can use to persist data
      * @param config the configuration features should use
+     * @param logger the logger instance to log disabled features
      * @return a collection of all features
      */
-    public static Collection<Feature> createFeatures(JDA jda, Database database, Config config) {
+    public static Collection<Feature> createFeatures(JDA jda, Database database, Config config,
+            Logger logger) {
```
Comment (Taz03) [inline application/src/main/java/org/togetherjava/tjbot/features/Features.java:82]: y don't u just create  a  logger in this class

File: application/src/main/java/org/togetherjava/tjbot/features/Features.java [hunk 2]
```diff
@@ -161,6 +165,11 @@ public static Collection<Feature> createFeatures(JDA jda, Database database, Con
         features.add(new JShellCommand(jshellEval));
 
         FeatureBlacklist<Class<?>> blacklist = blacklistConfig.normal();
-        return features.stream().filter(f -> blacklist.isEnabled(f.getClass())).toList();
+        return features.stream().filter(f -> !blacklist.isEnabled(f.getClass())).peek(f -> {
+            // Log INFO level message for each disabled feature using the provided logger
+            String message = "Feature '" + f.getClass().getSimpleName() + "' is disabled.";
+            logger.info(message);
+        }).toList();
```
Comment (Taz03) [inline application/src/main/java/org/togetherjava/tjbot/features/Features.java:172]: as u can see using peek is forbidden, so just put the log in filter method.

Also ur changing the logic before it removed the blacklisted features and kept all others, now it'll do the opposite

**Review by Taz03 — CHANGES_REQUESTED** [review]

**Comment by Taz03** [comment]
merged #1048

**Pull request closed without merging**